README.md

```diff
@@ -31,13 +89,182 @@ First, set up a container registry with:
+
+#### Setting up Workload Identity
+
+If you don't already have one, create a GCP service account to authorize the collector to send metrics, traces, and logs:
+
+```
+export GCLOUD_PROJECT=<your GCP project>
+export PROJECT_ID=<your Google Cloud project ID>
+gcloud iam service-accounts create otel-collector --project=${GCLOUD_PROJECT}
+```
+
+Give the service account access to write metrics, traces, and logs (or more/fewer roles based on your config):
+
+```
+gcloud projects add-iam-policy-binding $PROJECT_ID \
+    --member "serviceAccount:otel-collector@${GCLOUD_PROJECT}.iam.gserviceaccount.com" \
+    --role "roles/logging.logWriter"
+
+gcloud projects add-iam-policy-binding $PROJECT_ID \
+    --member "serviceAccount:otel-collector@${GCLOUD_PROJECT}.iam.gserviceaccount.com" \
+    --role "roles/cloudtrace.agent"
+    
+gcloud projects add-iam-policy-binding $PROJECT_ID \
+    --member "serviceAccount:otel-collector@${GCLOUD_PROJECT}.iam.gserviceaccount.com" \
+    --role "roles/monitoring.metricWriter"
+```
+
+Bind the service account to a Kubernetes ServiceAccount (you'll create this Kubernetes ServiceAccount later
+when you deploy the collector):
+
+```
+gcloud iam service-accounts add-iam-policy-binding "otel-collector@${GCLOUD_PROJECT}.iam.gserviceaccount.com" \
+    --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:${GCLOUD_PROJECT}.svc.id.goog[otel-collector/otel-collector]"
+```
+
+#### Grant Artifact Registry permissions
+
+If you are planning to build and host your collector image in Artifact Registry
+(see [Building with Cloud Build and Artifact Registry](#building-with-cloud-build-and-artifact-registry)),
+you will likely need to grant your GCP project's default service account with
+[Artifact Registry reader permissions](https://cloud.google.com/kubernetes-engine/docs/troubleshooting#permission_denied_error).
+
+Failure to do this may result in an `ImagePullBackOff` when trying to deploy the collector in GKE.
+
+You can find the name of this service account with `gcloud iam service-accounts list`:
+
+```
+$ gcloud iam service-accounts list
+DISPLAY NAME                            EMAIL                                                            DISABLED
+Compute Engine default service account  123456789123-compute@developer.gserviceaccount.com               False
+...
+```
+
+Give this service account reader permissions to the registry you plan to use for your collector images.
+For example, if you are using `gcr.io` in `us` with the account from above:
+
+```
+gcloud artifacts repositories add-iam-policy-binding gcr.io \
+    --location=us \
+    --member=123456789123-compute@developer.gserviceaccount.com \
+    --role="roles/artifactregistry.reader"
+```
+
+This concludes the prerequisites required to authenticate the OpenTelemetry collector to GCP
+using Workload Identity. Next, follow the steps below to deploy the Kubernetes manifests.
```

Comment:
note that artifact registry permissions aren't related to workload identity.  Also, if the registry is in the same project as your cluster, this permission already exists: https://cloud.google.com/artifact-registry/docs/integrate-gke#permissions.

Reply:
I confirmed that I didn't need this since my registry project and cluster project are the same

Reply:
This was not my experience on a new GCP account, and I spent a couple days debugging imagepullbackoffs before I found this to fix it. Maybe it was just something with my account, or this is a recent change, but these steps should at least be in a troubleshooting section

Reply:
Sounds reasonable.  Are you pushing the images to the same project as your cluster, or a different project?

Reply:
It was the same project as my cluster

Reply:
@dashpole is it possible that your service account already had these permissions? We should try this in a fresh project